### PR TITLE
fix: support hourly CDP uploads

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
https://github.com/ibi-group/otp-middleware/pull/170 added support for hourly CDP uploads. These uploads create filenames with a new format that was previously unsupported. While they appeared in the CDP file list, they weren't rendered correctly. This PR corrects this.

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]

**Review by binh-dam-ibigroup — COMMENTED** [review]
Could you advise regarding my comments about file names, please?

File: util/ui.ts [hunk 1]
```diff
@@ -94,9 +99,42 @@ export const getDateFromCDPFileName = (filename: string): string => {
...
+  const dateFormatter = new Intl.DateTimeFormat(
+    'en-US',
+    parsedDate.dateFormatterOptions
+  )
+  return dateFormatter.format(new Date(parsedDate.timestamp))
+}
+/**
+ * Some CDP file names are only a date. These are easy to parse
+ */
+const getDateOnlyFromCDPFileName = (date: string): FileNameParserResponse => {
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply util/ui.ts:115]: Yes `getDateOnlyFromCDPFileName` exists to provide compatibility with days-only zip files, especially as the hourly uploads may in future be combined into daily uploads.

I'm unable to replicate the issue you're facing with the January 2000 -- do you have a screenshot/link?

**Comment by miles-grant-ibigroup** [comment]
Thanks for the review, changes are addressed in 0bd1bcbfc23507a732df4c7ed3e2c4a0ddaedc12

File: util/ui.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply util/ui.ts:117]: image

Link: https://atlrides-otp-admin-ui-git-handle-new-cdp-filenames-ibi-group.vercel.app/?dashboard=cdp

**Comment by binh-dam-ibigroup** [comment]
@miles-grant-ibigroup Could you advise regarding https://github.com/ibi-group/otp-admin-ui/pull/71#discussion_r926790774?

File: util/ui.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply util/ui.ts:117]: Thank you for this! This looks to be a Blink-specific issue hence my not seeing it. Will look into it...

[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply util/ui.ts:117]: This is now fixed in 1d543ecf44c2f21173238205835b26e09b9dd3de

**Comment by miles-grant-ibigroup** [comment]
Thanks for catching that! That bug is now fixed

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks good now, thanks for the changes

File: util/ui.ts [hunk 2]
```diff
@@ -129,7 +129,8 @@ const getDateAndTimeFromCDPFileName = (
   date: string
 ): FileNameParserResponse => {
   const datePortion = date.split('-')
-  if (!datePortion) return null
+  if (!datePortion || datePortion.find((dp) => Number.isNaN(parseInt(dp))))
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply util/ui.ts:132]: Good catch thanks! Now that it's on one line I have not added brackets.

**Pull request merged**